src/main/java/by/itech/lab/supplier/domain/Application.java

```diff
@@ -3,114 +3,64 @@
 import lombok.AllArgsConstructor;
 import lombok.Builder;
 import lombok.Data;
+import lombok.EqualsAndHashCode;
 import lombok.NoArgsConstructor;
-
-import javax.persistence.*;
-import java.sql.Date;
+import org.hibernate.annotations.Where;
+
+import javax.persistence.CascadeType;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.EnumType;
+import javax.persistence.Enumerated;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.OneToMany;
+import javax.persistence.Table;
+import java.time.LocalDate;
+import java.util.HashSet;
+import java.util.Set;
 
 @Data
 @AllArgsConstructor
 @NoArgsConstructor
 @Builder
 @Entity
-@Table(name = "application")
-public class Application {
-
-    private Long id;
-    private String number;
-    private Date registrationDate;
-    private Date lastUpdated;
-    private Address destinationLocationAddress;
-    private User createdByUsers;
-    private User lastUpdatedByUsers;
-    private ApplicationStatus applicationStatus;
-    private WayBill wayBill;
+@Table
+@Where(clause = "deleted_at is null")
+public class Application implements BaseEntity {
 
     @Id
     @GeneratedValue(strategy = GenerationType.IDENTITY)
-    public Long getId() {
-        return id;
-    }
-
-    public void setId(Long id) {
-        this.id = id;
-    }
-
-    @Column(name = "number")
-    public String getNumber() {
-        return number;
-    }
-
-    public void setNumber(String number) {
-        this.number = number;
-    }
-
-    @Column(name = "registration_date")
-    public Date getRegistrationDate() {
-        return registrationDate;
-    }
-
-    public void setRegistrationDate(Date registrationDate) {
-        this.registrationDate = registrationDate;
-    }
-
-    @Column(name = "last_updated")
-    public Date getLastUpdated() {
-        return lastUpdated;
-    }
-
-    public void setLastUpdated(Date lastUpdated) {
-        this.lastUpdated = lastUpdated;
-    }
-
+    private Long id;
+    @Column(unique = true, nullable = false)
+    private String number;
+    @Column(nullable = false)
+    private LocalDate registrationDate;
+    @Column(nullable = false)
+    private LocalDate lastUpdated;
     @ManyToOne
     @JoinColumn(name = "source_location_address_id")
-    public Address getDestinationLocationAddress() {
-        return destinationLocationAddress;
-    }
-
-    public void setDestinationLocationAddress(Address destinationLocationAddress) {
-        this.destinationLocationAddress = destinationLocationAddress;
-    }
-
+    private Address sourceLocationAddressId;
+    @ManyToOne
+    @JoinColumn(name = "destination_location_address_id")
+    private Address destinationLocationAddressId;
```

Comment:
Could be named as destinationLocationAddress

Reply:
+

Reply:
+++